jobs/job_utils.py

```diff
@@ -184,6 +211,12 @@ def get_model_from_beam_entity(beam_entity):
 def get_beam_query_from_ndb_query(query):
     """Returns an equivalent Apache Beam query from the given NDB query.
 
+    This function helps developers avoid learning two types of query syntaxes.
+    Specifically, the datastoreio module offered by the Apache Beam SDK only
+    accepts Beam datastore queries, and are implemented very differently from
+    NDB queries. This function adapts the two patterns to make job code easier
+    to write.
```

Comment:
Addresses https://github.com/oppia/oppia/pull/12880#discussion_r638269496:
> Hmm, bit unsure if we really need this function? Do we expect many cases when we would call `get_model_key` with something else than `datastore_services.Model`? Because I would say that in a code where I know that I have a `datastore_services.Model` I would use `key` directly.